src/compile/selection/assemble.ts

```diff
@@ -114,23 +114,29 @@ export function assembleTopLevelSignals(model: UnitModel, signals: Signal[]) {
 
 export function assembleUnitSelectionData(model: UnitModel, data: readonly VgData[]): VgData[] {
   const dataCopy = [...data];
+  const unit = unitName(model, {escape: false});
+
   for (const selCmpt of vals(model.component.selection ?? {})) {
-    const init: VgData = {name: selCmpt.name + STORE};
+    const store: VgData = {name: selCmpt.name + STORE};
+
+    if (selCmpt.project.hasSelectionId) {
+      store.transform = [{type: 'collect', sort: {field: SELECTION_ID}}];
```

Comment:
@arvind regarding https://github.com/vega/vega-lite/pull/7932 and this PR I have a question:
- Is the transform for datum.values[0] missing here? because you mentioned that the transform for the lasso must include this formula to correctly work with 'vlSelectionTuples'

`
"transform": [
          {"type": "formula", "expr": "datum.values[0]", "as": "_vgsid_"},
          {"type": "collect", "sort": {"field": "_vgsid_"}}
        ]
`

Reply:
Yes, good eye! In https://github.com/vega/vega-lite/pull/7933 and https://github.com/vega/vega/pull/3397 (both of which this PR depends on), I introduced a more optimized format for selection stores when only the `SELECTION_ID` is being stored. In particular, after those two PRs, the selection store holds tuples in the form `{_vgsid_, unit}` sorted by `_vgsid_`. This structure allows us to skip the `formula` transform, as well all the array generation that would previously happen for the `fields` and `values` keys of the older tuples. The `{unit, fields, values}` tuple format still applied for non-selectionID based stores. 

Hope that helps!